web/modules/custom/hoeringsportal_deskpro/src/Service/HearingHelper.php

```diff
@@ -447,20 +447,44 @@ public function runSynchronizeTicket(array $payload) {
       throw new \RuntimeException($message);
     }
 
-    $this->logger->debug(
-      NULL === $ticketIndex
-      ? sprintf('Adding ticket %s on hearing %s', $ticketId, $hearing->id())
-      : sprintf('Updating ticket %s on hearing %s (index: %d)', $ticketId, $hearing->id(), $ticketIndex),
-      [
-        '@ticket' => json_encode($ticket),
-      ]
-    );
-
-    if (NULL === $ticketIndex) {
-      $tickets[] = $ticket;
+    // Remove unpublished or deleted tickets.
+    $remove =
+      // Ticket is unpublished if and only if the field "unpublish_reply" === 1.
+      1 === (int) ($ticket['fields']['unpublish_reply'] ?? NULL)
+      // A deleted ticket has a status starting with "hidden" (and is not really
+      // deleted so it makes sense to get it from the api).
+      || 0 === strpos($ticket['status'], 'hidden');
```

Comment:
Are we still in PHP 7.x or `str_starts_with` could be used.

Reply:
We're still on php 7.4 (but hopefully not for long).